### PR TITLE
tx trace api

### PR DESCRIPTION
**Pull request by zhangsoledad**

### CODE REVIEW

**Comment by doitian** [comment]
bors try

File: pool/src/txs_pool/trace.rs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        TxTraceMap {
+            inner: LruCache::new(capacity),
+        }
+    }
+
+    #[allow(clippy::needless_pass_by_value)]
+    pub fn add_pending<S: ToString>(&mut self, hash: &H256, info: S) {
+        self.inner
+            .entry(hash.clone())
+            .or_insert_with(Vec::new)
```
Comment (quake) [inline pool/src/txs_pool/trace.rs]:
```suggestion
            .or_default()
```

**Comment by doitian** [comment]
Travis link to the checks seems broken.

https://travis-ci.com/nervosnetwork/ckb/builds/97198673

File: rpc/doc.md [hunk 2]
```diff
@@ -332,3 +332,77 @@ curl -d '{"id": 2, "jsonrpc": "2.0", "method":"send_transaction","params": [{"ve
...
+}
+```
+
+# get_transaction_trace
+
+Creates new transaction.
+
+## Parameters
+
+Transaction - The transaction object.
```
Comment (doitian) [inline rpc/doc.md]: Copy-paste error

**Comment by zhangsoledad** [comment]
@doitian  fixed

**Pull request merged**